### PR TITLE
(maint) Fix failing Pester tests

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Fix failing CI tests
- Improve error reporting when they fail
- Small fix into tab completion for ensuring expected & handled errors don't pollute $error for users. 
  - This was actually the cause of some of the failing tests, as they were checking $error to determine if something failed or not. :(

## Motivation and Context

Wanting to sort out our test failures before release.

## Testing

Tested locally where I can, but I will also be testing in CI.

### Operating Systems Testing
Win11

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.
* [x] Pester tests updated

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [x] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

N/A, tests only.

### CODE REVIEW

File: tests/chocolatey-tests/commands/choco-upgrade.Tests.ps1 [hunk 1]
```diff
@@ -272,7 +272,10 @@ To upgrade a local, or remote file, you may use:
             $Output = Invoke-Choco upgrade upgradepackage --confirm
         }
 
-        It 'Exits with Success (0)' {
+
+        # `upgradepackage` contains a beforeModify that throws, which triggers an incorrect -1 exit code.
+        # See https://app.clickup.com/t/20540031/PROJ-615
+        It 'Exits with Success (0)' -Pending {
```
Comment (AdmiringWorm) [inline tests/chocolatey-tests/commands/choco-upgrade.Tests.ps1:278]: Don't mark it as Pending, but rather with
```suggestion
        It 'Exits with Success (0)' -Tag Broken {

```

**Review by AdmiringWorm — DISMISSED** [review]

File: tests/chocolatey-tests/commands/choco-upgrade.Tests.ps1 (hunk 1 above)
Comment (vexx32) [reply tests/chocolatey-tests/commands/choco-upgrade.Tests.ps1]: Thanks! Fixed :)

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@vexx32 thanks for getting these fixed up!

**Pull request merged**